docs/cip/README.md

```diff
@@ -461,247 +1988,671 @@ The Leios protocol may have to mitigate the following categories of threats.
+
+[ripe-atlas]: https://atlas.ripe.net/ "RIPE Atlas"
+[cardano-foundation]: https://cardanofoundation.org/ "Cardano Foundation"
+
+<!-- License -->
+
+[apache-2.0]: http://www.apache.org/licenses/LICENSE-2.0 "Apache License 2.0"
+
+<!-- Footnotes -->
+
+[^fasort]: The Fait Accompli sortition scheme
+
+[^2]: Leios: Dynamic Availability for Blockchain Sharding (2025)
+
+[^leioscrypto]: Leios cryptography prototype implementation
+
+## Appendix
+
+<h3 id="appendix-a-requirements">Appendix A: Requirements for votes and certificates</h2>
+
+The voting and certificate scheme for Leios must satisfy the following
+requirements to ensure security, efficiency, and practical deployability:
+
+1. **Succinct registration of keys:** The registration of voting keys should not
+   involve excessive data transfer or coordination between parties. Ideally,
+   such registration would occur as part of the already existing operational
+   certificates and not unduly increase their size.
+
+2. **Key rotation:** The cryptographic keys used to sign Leios votes and
+   certificates _do not_ need to be rotated periodically because the constraints
+   on Leios voting rounds and the key rotation already present in Praos secure
+   the protocol against attacks such as replay and key compromise.
+
+3. **Deterministic signatures:** Deterministic signatures can guard against
+   attacks that weaken key security.
+
+4. **Local sortition:** Selection of the voting committee should not be so
+   deterministic and public as to open attack avenues such as denial-of-service
+   or subversion.
+
+5. **Liveness:** Adversaries with significant stake (e.g., more than 35%) should
+   not be able to thwart an honest majority from reaching a quorum of votes for
+   an EB.
+
+6. **Soundness:** Adversaries with near majority stake (e.g., 49%) should not be
+   able to form an adversarial quorum that certifies the EB of their choice.
+
+7. **Small votes:** Because vote traffic is large and frequent in Leios, the
+   votes themselves should be small. Note that the large size of Praos KES
+   signatures precludes their use for signing Leios votes.
+
+8. **Small certificates:** Because Leios certificates are frequent and must fit
+   inside Praos blocks, they should be small enough so there is plenty of room
+   for other transactions in the Praos blocks. Note once again that certificates
+   based on Praos KES signatures are too large for consideration in Leios.
+
+9. **Fast cryptography:** The computational burden of creating and verifying
+   voting eligibility, the votes themselves, and the resulting certificate must
+   be small enough to fit within the CPU budget for Leios stages.
+
+The aggregate signature scheme implementation specified in this document (using
+BLS as an example) satisfies all these requirements, as evidenced by the
+performance characteristics and certificate sizes documented in the
+[Specification for votes and certificates](#specification-for-votes-and-certificates)
```

Comment:
I think this link is dead as the corresponding section was removed and moved to a different folder/file.